web/src/app/users/linkToMobile/LinkToMobile.tsx

```diff
@@ -0,0 +1,202 @@
+  return {
+    cancelButton,
+    dialog: {
+      height: 330,
+    },
+    successContainer: {
+      height: '100%',
+      width: '100%',
+      display: 'flex',
+      alignItems: 'center',
+      textAlign: 'center',
+    },
+  }
+})
+
+const mutation = gql`
+  mutation {
+    createAuthLink {
+      id
+      claimCode
+    }
+  }
+`
+export const query = gql`
+  query authLinkStatus($id: ID!) {
+    authLinkStatus(id: $id) {
+      id
+      expiresAt
+      claimed
+      verified
+      authed
+    }
+  }
+`
+
+export default function LinkToMobile(): JSX.Element {
+  const classes = useStyles()
+  const width = useWidth()
+  const fullscreen = isWidthDown('md', width)
+  const [showDialog, setShowDialog] = useState(false)
+  const [index, setIndex] = useState(0)
+
+  const [createAuthLink, createAuthLinkStatus] = useMutation(mutation)
+  const loading = !createAuthLinkStatus.data && createAuthLinkStatus.loading
+  const authLinkID = createAuthLinkStatus?.data?.createAuthLink.id ?? ''
+  const claimCode = createAuthLinkStatus?.data?.createAuthLink.claimCode ?? ''
+
+  const { data } = useQuery(query, {
+    variables: {
+      id: authLinkID,
+    },
+    skip: loading || !authLinkID,
+  })
+
+  const claimed = data?.authLinkStatus.claimed ?? ''
+  const authed = data?.authLinkStatus.authed ?? ''
+
+  useEffect(() => {
+    if (showDialog) {
+      createAuthLink()
+    }
+  }, [showDialog])
+
+  useEffect(() => {
+    if (claimed) {
+      setIndex(1)
+    }
+  }, [claimed])
+
+  useEffect(() => {
+    if (authed) {
+      setIndex(2)
+    }
+  }, [authed])
+
+  function slideRenderer({ index, key }: SlideParams): ReactNode {
+    switch (index) {
+      case 0:
+        return (
+          <ClaimCodeDisplay
+            key={key}
+            authLinkID={authLinkID}
+            claimCode={claimCode}
+          />
+        )
+      case 1:
+        return <VerifyCodeFields key={key} authLinkID={authLinkID} />
+      case 2:
+        return <Success key={key} isStopped={!authed} />
+      case 3:
+        return <Retry key={key} />
+      default:
+        return null
+    }
+  }
+
+  return (
+    <React.Fragment>
+      <Button
+        color='primary'
+        variant='contained'
+        startIcon={<PhonelinkIcon />}
+        onClick={() => setShowDialog(true)}
+      >
+        Link to Mobile
+      </Button>
+
+      <Dialog
+        classes={{ paper: classes.dialog }}
+        open={showDialog}
+        fullScreen={fullscreen}
+        fullWidth
+        maxWidth='xs'
+        onClose={() => setShowDialog(false)}
+      >
+        <DialogTitleWrapper title='Link to Mobile' />
+        {loading ? (
+          <DialogContent>
+            <Spinner />
+          </DialogContent>
+        ) : (
+          <VirtualizeAnimatedViews
+            index={index}
+            onChangeIndex={(i: number) => setIndex(i)}
+            slideRenderer={slideRenderer}
+          />
+        )}
+        <DialogActions>
+          {index === 2 ? (
```

Comment:
Maybe we should store this in a variable, or a constant to compare with so we don't need context to understand the ternary.

```js
index === lastStepIndex
```

or

```js
isDone ? (
```